### DISCUSSION TITLE
Can I read all KV v1 secrets once?

### DISCUSSION THREAD

**Discussion by matanmarciano** [hvac/hvac#926] [Q&A]
Hey there,
Can I read all KV v1 secrets once?

Currently, I get the list and loop on the keys to read each secret:
```
    existingSecretsKvv1 = dict()
    keys = client.secrets.kv.v1.list_secrets(path='myPath')['data']['keys']
    for key in keys:
        secret_value = client.secrets.kv.v1.read_secret(path='myPath/{}'.format(key))
        existingSecretsKvv1[key] = secret_value['data']['value']

    print(existingSecretsKvv1)
```

But it is too slow to read them one by one. 
Any chance I can get all keys (AND values) at once?

**Comment by briantist** [comment]
Hi @matanmarciano , there is no way to do that with Vault itself, so as such there is no way to do that in `hvac`.

Parallelizing the reads may be the way to go as long as your Vault server can handle it.

From a workflow design POV, I might wonder why you would need to read a large number of secrets all at once, and in a short amount of time.

I might have recommended using a Vault Agent to pull and cache your secrets in the background but unfortunately they still don't support caching KV secrets so 😔
